internal/driver/controllerserver_helper.go

```diff
@@ -167,19 +167,27 @@ func (cs *ControllerServer) getContentSourceVolume(ctx context.Context, contentS
 		return nil, errInternal("source volume *linodego.Volume is nil") // Throw an internal error if the processed linodego.Volume is nil
 	}
 
-	// Check if the volume's region matches the server's metadata region
-	if volumeData.Region != cs.metadata.Region {
+	// Check if the source volume's region matches the server's metadata region
+	// If no topology is specified, the source volume must be in the same region as the server's metadata region
+	if accessibilityRequirements == nil && volumeData.Region != cs.metadata.Region {
 		return nil, errRegionMismatch(volumeData.Region, cs.metadata.Region)
 	}
 
+	// If a topology is specified, the source volume must be in the same region as the specified topology
+	if accessibilityRequirements != nil {
+		if volumeData.Region != accessibilityRequirements.GetPreferred()[0].GetSegments()[VolumeTopologyRegion] {
```

Comment:
is this guaranteed to be an array of len > 0  && is the key guaranteed to exist?

Reply:
Its pretty much guaranteed to exist and the first key is going to be the region we want to use.

for more info check this link out! https://kubernetes-csi.github.io/docs/topology.html?highlight=VOLUME_ACCESSIBILITY_CONSTRAINTS#implementing-topology-in-your-csi-driver

Reply:
I was curious because `getRegionFromTopology` operates on the same accessibilityRequirements but includes all those checks, Thanks for the link - will check it out!

Reply:
could also use that here - to validate

Reply:
The only time that is not passed (accessibilityRequirements) is when you have a broken multi region cluster (I had this issue while trying to make konnectivity work). In that senario, I'm falling back to using the region from the metadata (region where controller server is running)

Reply:
Good catch. Let me include those checks here as well

Reply:
Updated it!